content/purgatorio/launch-and-mantain-the-system/monitoring-usage-and-behavior.md

```diff
@@ -1,5 +1,142 @@
 ---
 title: Monitoring Usage and Behavior
 author: clone95
-description: ---
----
\ No newline at end of file
+description: This guide introduces you to the best practices of monitoring Machine Learning models in production, and ensure that they behave as expected.
+---
+
+# Index 
+
+- [Trusting is Good Not Trusting is Better](#Trusting-is-Good-Not-Trusting-is-Better)
+- [Monitoring the Behavior](#Monitoring-the-Behavior)
+- [Monitoring the Usage](#Monitoring-the-Usage)
+- [Conclusions](#Conclusions)
+
+
+## Trusting is Good Not Trusting is Better
+
+Application monitoring is a _key_ part of running software in production. 
+
+Without it, the only way of finding an issue is through sheer luck or because a client has reported it. 
+
+Both of which are less than ideal, to say the least!
+
+**You wouldn't deploy an application without monitoring, so why do it for Machine Learning models?**
+
+So let's Start!
+
+The first resource you should go through is the following: 
+
+- [How to Monitor Machine Learning Models](https://christophergs.com/machine%20learning/2020/03/14/how-to-monitor-machine-learning-models/)
+
+It's a very detailed and comprehensive blog post and it addresses these topics:
+
+- What makes ML systems monitoring hard?
+- How can we monitor the usage and behavior of the model? 
+- What are the key metrics to track and alert on?
+- What are the key principles for monitoring the ML system?
+
+Once you are done with the previous blog post, you can refer to the related chapter of the book [Machine Learning Engineering](http://www.mlebook.com/wiki/doku.php) for a more detailed guide (strongly recommended buy, but you can read it for free online).
+
+- [Chapter 8 - Model Serving and Monitoring](https://www.dropbox.com/s/nj23m5myng9gram/Chapter8.pdf?dl=0)
+
+See also:
+ - [There are two very different ways to deploy ML models, here’s both](https://towardsdatascience.com/there-are-two-very-different-ways-to-deploy-ml-models-heres-both-ce2e97c7b9b1)
+
+With these two resources, you should understand broadly the reasons and challenges of monitoring the Machine Learning models in production, and you should have reasonable strategies to tackle them. 
+
+As you know, the two main challenges regarding the model monitoring are:
+
```

Comment:
One thing to mention here is to always have a baseline before we start monitoring or measuring, if we are starting for the first time then that is our baseline.

So for the situations application, an extra first step is establish a baseline and keep that static and make all comparisons and references with regards to this baseline.

- How has the system been behaving since [important milestone/change]? I hope here runtime and model performance is also considered behaviour?

Reply:
Nice insightful idea, inserted a subsection for that :D

Reply:
@clone95 I noticed there were issues with my phrases so I have corrected/changed them to reflect my ideas, pls have a re-read to see if you like it still and apply this version on top of your inserted ones.